protected.go

```diff
@@ -53,14 +53,21 @@ func (addr *protectedAddr) TCPAddr() *net.TCPAddr {
 }
 
 // New construct a protector from the protect function and DNS server IP address.
-func New(protect Protect, dnsServerIP string) *Protector {
-	ipAddr := net.ParseIP(dnsServerIP)
+func New(protect Protect, dnsServer string) *Protector {
```

Comment:
With these changes, we can pass in the full dnsgrab server address from [here](https://github.com/getlantern/lantern-build/blob/update-protected/lantern-mobile/app/src/main/assets/settings.yaml#L3). If the address is invalid for whatever reason, we default to using `8.8.8.8:53`

Reply:
So the default is "0.0.0.0:53" correct @atavism? That makes sense as an address to bind to locally (so it's grab any address), but are we actually running a local DNS server now? If not I'm a little confused!

Reply:
@myleshorton No, sorry, `8.8.8.8:53` is only the default if we pass in an address to protected that's invalid. Yes, we're running a local DNS server: https://github.com/getlantern/dnsgrab/blob/master/dnsgrab.go